CRM/Utils/Hook.php

```diff
@@ -1635,6 +1635,23 @@ public static function disable() {
     );
   }
 
+  /**
+   * Alter redirect.
+   *
+   * This hook is called when the browser is being re-directed and allows the url
+   * to be altered.
+   *
+   * @param string $url
+   *
+   * @return mixed
+   */
+  public static function alterRedirect($url) {
```

Comment:
Shouldn't the parameter be alterable?  To my reading, a listener who prefers to alter the URL would need to execute the redirect themselves (e.g. emitting the `Location:` header and `exit()`ing).

If you read the code in `CRM_Utils_System::redirect()`, there's actually a bit nuance to how the redirect is executed. (Ex: An AJAX snippet works different from a standalone page. Ex: it's not `exit()`; it's `civiExit()`.) The hook consumer should be focused on the content of the URL -- not on the mechanism of delivering that content.

Reply:
I changed it to &$url